cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -67,6 +70,15 @@
   private static final Type SET_METADATA_RECORD_TYPE = new TypeToken<Set<MetadataRecord>>() { }.getType();
   private static final Type SET_METADATA_SEARCH_RESULT_TYPE =
     new TypeToken<Set<MetadataSearchResultRecord>>() { }.getType();
+
+  private static final Function<String, MetadataSearchTargetType> STRING_TO_TARGET_TYPE =
+    new Function<String, MetadataSearchTargetType>() {
+      @Override
+      public MetadataSearchTargetType apply(String input) {
+        return MetadataSearchTargetType.valueOf(input.toUpperCase());
```

Comment:
`valueOfSerializedForm` is used during serialization, in this case we need `valueOf()`